### DISCUSSION TITLE
JDK 18, 19, 20 support?

### DISCUSSION THREAD

**Discussion by selundqma** [eclipse-tycho/tycho#1253] [Q&A] [answered]
Hi!
Just a small question. Will Tycho also support JDK 18, 19, 20 (non-LTS version)?
I have tried to update our projects to JDK 18 by changing to "JavaSE-18" in the files, but I get build error:

`[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:2.7.4:compile (default-compile) on project com.selundmq.cs.osgi: Fatal error compiling: target level should be in '1.1'...'1.8','9'...'17' (or '5.0'..'17.0') or cldc1.1: 18 -> [Help 1]`

**Comment by laeubi** [comment, marked as answer]
Tycho itself support any java version, but the **compiler(!)** that is [JDT](https://github.com/eclipse-jdt/eclipse.jdt.core) might not support all versions so you need either use `javac` as compiler backend or wait for [JDT](https://github.com/eclipse-jdt/eclipse.jdt.core) to release a never versions of Java support.
